frontend/components/user-settings/user-settings.tsx

```diff
@@ -0,0 +1,298 @@
+        return <div>Loading user data...</div>
+    }
+
+    if (error) {
+        return <div>Error: Failed to load user data</div>
+    }
+
+    if (!user) {
+        return <div>No user data available.</div>
+    }
+
+    return (
+        <div className="container mx-auto p-4">
+        <h1 className="text-2xl font-bold mb-4">User Settings for {originalUsername}</h1>
+        <Tabs defaultValue="profile">
+            <TabsList className="mb-4">
+                <TabsTrigger value="profile">Profile</TabsTrigger>
+                <TabsTrigger value="account">Account</TabsTrigger>
+            </TabsList>
+            <TabsContent value="profile">
+            <Card>
+                <CardHeader>
+                    <CardTitle>Profile Information</CardTitle>
+                    <CardDescription>Update your profile details here.</CardDescription>
+                </CardHeader>
+                <CardContent className="space-y-4">
+                    <div className="flex items-center space-x-4">
+                        <Avatar className="w-24 h-24">
+                        <AvatarImage src={profilePicture} alt="Profile picture" />
+                        <AvatarFallback>Placeholder</AvatarFallback>
+                        </Avatar>
+                        <div>
+                        <Input type="file" accept="image/*" onChange={handleProfilePictureChange} className="mb-2" />
+                        <Button variant="outline" onClick={handleDeleteProfilePicture}>Delete Picture</Button>
+                        </div>
+                    </div>
+                    <div>
+                        <Label htmlFor="username">Username</Label>
+                        <Input id="username" name="username" value={user.username} onChange={handleInputChange} />
+                    </div>
+                    <div>
+                        <Label htmlFor="email">Email</Label>
+                        <Input id="email" name="email" type="email" value={user.email} onChange={handleInputChange} />
+                    </div>
+                    <div>
+                        <Label htmlFor="skillLevel">Skill Level</Label>
+                        <Select value={user.skillLevel} onValueChange={handleSkillLevelChange}>
+                        <SelectTrigger id="skillLevel">
+                            <SelectValue placeholder="Select skill level" />
+                        </SelectTrigger>
+                        <SelectContent>
+                            <SelectItem value="Novice">Novice</SelectItem>
+                            <SelectItem value="Intermediate">Intermediate</SelectItem>
+                            <SelectItem value="Expert">Expert</SelectItem>
+                        </SelectContent>
+                        </Select>
+                    </div>
+                </CardContent>
+                <CardFooter>
+                <Button onClick={handleSaveChanges}>Save Changes</Button>
+                <Toaster />
```

Comment:
Oh I just realised that the toaster is put here, I think it should be placed in `layout.tsx`
![Uploading image.png…]()


https://ui.shadcn.com/docs/components/toast#installation

Reply:
Will shift this to the `layout.tsx` file.